src/libexpr/eval.cc

```diff
@@ -1488,8 +1507,12 @@ void EvalState::callFunction(Value & fun, size_t nrArgs, Value * * args, Value &
                heap-allocate a copy and use that instead. */
             Value * args2[] = {allocValue(), args[0]};
             *args2[0] = vCur;
-            /* !!! Should we use the attr pos here? */
-            callFunction(*functor->value, 2, args2, vCur, pos);
+            try {
+                callFunction(*functor->value, 2, args2, vCur, *functor->pos);
+            } catch (Error & e) {
+                e.addTrace(pos, "While calling a functor (an attribute set with a 'functor' attribute)");
```

Comment:
Right. But I think the error message should explain that, because it may not be obvious to a novice user following the trace why an attrset gets applied. These traces are not intended for expert users most of the time. Especially these basic type/usage errors that novices are more prone to do. This is a small atempt at making nix more discoverable by providing context when the error appears.